peers/validators/canonical_validator_client.go

```diff
@@ -59,15 +66,27 @@ func (v *CanonicalValidatorClient) GetCurrentCanonicalValidatorSet(subnetID ids.
 }
 
 func (v *CanonicalValidatorClient) GetMinimumHeight(ctx context.Context) (uint64, error) {
-	return v.client.GetHeight(ctx)
+	return v.client.GetHeight(ctx, v.options...)
 }
 
 func (v *CanonicalValidatorClient) GetCurrentHeight(ctx context.Context) (uint64, error) {
-	return v.client.GetHeight(ctx)
+	return v.client.GetHeight(ctx, v.options...)
 }
 
 func (v *CanonicalValidatorClient) GetSubnetID(ctx context.Context, blockchainID ids.ID) (ids.ID, error) {
-	return v.client.ValidatedBy(ctx, blockchainID)
+	return v.client.ValidatedBy(ctx, blockchainID, v.options...)
+}
+
+func (v *CanonicalValidatorClient) GetCurrentValidators(ctx context.Context, subnetID ids.ID, nodeIDs []ids.NodeID) ([]platformvm.ClientPermissionlessValidator, error) {
+	return v.client.GetCurrentValidators(ctx, subnetID, nodeIDs, v.options...)
+}
+
+func (v *CanonicalValidatorClient) GetValidatorsAt(
+	ctx context.Context,
+	subnetID ids.ID,
+	height uint64,
+) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+	return v.client.GetValidatorsAt(ctx, subnetID, height, v.options...)
```

Comment:
Is there a reason why it's necessary to export these methods? My worry is there are:
- `GetCurrentCanonicalValidatorSet`
- `GetValidatorsAt`
- `GetCurrentValidators`
- `getCurrentValidatorSet`
- `GetValidatorSet`

Which all do slightly different things but make it confusing to know which should be used. If only called internally here, I think it'd prefer to directly call `v.client.GetValidatorsAt` and `v.client.GetCurrentValidators` in the existing implementations rather than add additional similarly named wrapped.

Reply:
`CanonicalValidatorClient` needs to implement `warp.ValidatorState` in order to be able to call `GetCanonicalValidatorSet`. It looks like that interface has been updated in AvalancheGo since this was last updated here, so `GetValidatorSet` should be the only method we need to export.